### ISSUE TITLE
Extended Tests Use `.` Imports

### ISSUE THREAD

**Issue by stevekuznetsov** [openshift/origin#4268]
We need to remove all `import . "path/to/package"` imports from the `test/extended` files. This import scheme is hard to read and understand, and furthermore breaks any go-to-definition tooling. Current offenders:

```
test/extended/util/cli.go:22:	. "k8s.io/kubernetes/test/e2e"
test/extended/default_builds_test.go:8:	. "github.com/onsi/ginkgo"
test/extended/default_builds_test.go:9:	. "github.com/onsi/gomega"
test/extended/default_builds_test.go:10:	. "k8s.io/kubernetes/test/e2e"
test/extended/default_mysql_test.go:7:	. "github.com/onsi/ginkgo"
test/extended/default_mysql_test.go:8:	. "github.com/onsi/gomega"
```

**Comment by stevekuznetsov** [comment]
@mfojtik Keep this in mind if you're making further changes - let's not introduce more.

**Comment by liggitt** [comment]
+1

**Comment by mfojtik** [comment]
@stevekuznetsov http://onsi.github.io/ginkgo/#avoiding-dot-imports

We will have to redefine the ginkgo helpers ^^ in order to do this. However I don't think we can avoid dot import for upstream e2e.

**Comment by mfojtik** [comment]
@stevekuznetsov another problem is that we will have to remap Describe/It/By in each extended test package, which will be pain and will look ugly.

**Comment by stevekuznetsov** [comment]
@mfojtik Looking nice falls pretty close to last behind usability, readability, etc. I don't think prepending the package names before function calls (as we do in the rest of the code base, right?) will be bad. To be honest comparing the extended test framework we have with our integration tests is difficult. Writing an integration test is fairly simple and straightforward. Writing an extended test, no so much. The first thing I do when I look at code is I try to find the definitions of functions being called. If for no other reason than to allow tooling to pick up function definition locations, we need this change. A file where  I have to grep the entire repo to find potential places where the functions are defined is atrocious.

**Comment by mfojtik** [comment]
@stevekuznetsov I understand why we should avoid dot imports, just saying that it is not that trivial as it looks like. For example, kubernetes e2e use the ginkgo and gomega helpers with dot import. If I call the function from kubernetes e2e that call the gomega/ginkgo helper using dot import, will it work? Or do we need to change upstream e2e to also avoid dot imports?

Is the go-to-definition suite broken completely if the go file has dot import or just the go-to-definition of the ginkgo helpers is broken?

**Comment by stevekuznetsov** [comment]
I see. It may be hard but I'm not against trying.

Dot-imported function calls don't seem to be able to be followed by any go-to-definition tool I have. The other definitions in the file are fine.

**Comment by deads2k** [comment]
I think that is valuable to know the pedigree of the method you're calling at a glance.  Using a `.` to hide the origin of a method is jarring during a scan.  

I don't find "it looks prettier" to be compelling enough to smash it into the current package's namespace.

**Comment by mfojtik** [comment]
@deads2k agreed, we still have to investigate the upstream e2e helpers, but I'm OK prepending helpers with the package name (we can make it shorter ginkgo -> g, gomega -> go)

**Comment by deads2k** [comment]
> @deads2k agreed, we still have to investigate the upstream e2e helpers, but I'm OK prepending helpers with the package name (we can make it shorter ginkgo -> g, gomega -> gm)

I have no issue with assigning a different qualifier to the import.

**Comment by mfojtik** [comment]
Fixed here: https://github.com/openshift/origin/pull/4285

**Comment by jhadvig** [comment]
@mfojtik close pls

**Comment by stevekuznetsov** [comment]
Fixed in #4285
